ocaml-lsp-server/src/ocaml_lsp_server.ml

```diff
@@ -336,7 +336,20 @@ let hover (state : State.t) { HoverParams.textDocument = { uri }; position; _ }
   | None -> Fiber.return None
   | Some (loc, typ) ->
     let syntax = Document.syntax doc in
-    let+ doc = query_doc doc pos in
+    let+ doc = query_doc doc pos
+    and+ typ =
+      (* We ask Ocamlformat to format this type *)
+      let+ result = Ocamlformat_rpc.query_type state.ocamlformat_rpc typ in
+      match result with
+      | Ok v -> v
+      | Error `No_process -> typ
+      | Error (`Msg message) ->
+        (* We log OCamlformat errors and display the unformated type *)
+        Log.log ~section:"ocamlformat" (fun () ->
+            Log.msg "An error occured while querying ocamlformat"
```

Comment:
Note that this log is mostly used for debugging. Users will not see these messages unless they open the debug pane. If you want users to see this error, use the `ShowMessage` notification.